api/client/credentials.go

```diff
@@ -17,84 +17,98 @@ limitations under the License.
+// also provide other functionality, such as automatic address discovery and
+// ssh connectivity.
+//
+// See the examples below for an example of each loader.
 type Credentials interface {
-	// Dialer is used to create a dialer used to connect to Auth.
+	// Dialer is used to create a dialer used to connect to the Auth server.
 	Dialer(cfg Config) (ContextDialer, error)
-	// TLSConfig returns TLS configuration used to connect to Auth.
+	// TLSConfig returns TLS configuration used to authenticate the client.
 	TLSConfig() (*tls.Config, error)
-	// SSHClientConfig returns SSH configuration used to connect to Proxy through tunnel.
+	// SSHClientConfig returns SSH configuration used to connect to the
+	// Auth server through a reverse tunnel.
 	SSHClientConfig() (*ssh.ClientConfig, error)
 }
 
-// LoadTLS is used to load credentials directly from another *tls.Config.
-func LoadTLS(tlsConfig *tls.Config) *TLSConfigCreds {
-	return &TLSConfigCreds{
+// LoadTLS is used to load Credentials directly from a *tls.Config.
+//
+// TLS creds can only be used to connect directly to a local Teleport Auth server.
+func LoadTLS(tlsConfig *tls.Config) Credentials {
+	return &tlsConfigCreds{
 		tlsConfig: tlsConfig,
 	}
 }
 
-// TLSConfigCreds are used to authenticate the client
-// with a predefined *tls.Config.
-type TLSConfigCreds struct {
+// tlsConfigCreds use a defined *tls.Config to provide client credentials.
+type tlsConfigCreds struct {
 	tlsConfig *tls.Config
 }
 
-// Dialer is used to dial a connection to Auth.
-func (c *TLSConfigCreds) Dialer(cfg Config) (ContextDialer, error) {
+// Dialer is used to dial a connection to an Auth server.
+func (c *tlsConfigCreds) Dialer(cfg Config) (ContextDialer, error) {
 	return nil, trace.NotImplemented("no dialer")
 }
 
-// TLSConfig returns TLS configuration used to connect to Auth.
-func (c *TLSConfigCreds) TLSConfig() (*tls.Config, error) {
+// TLSConfig returns TLS configuration.
+func (c *tlsConfigCreds) TLSConfig() (*tls.Config, error) {
 	if c.tlsConfig == nil {
 		return nil, trace.BadParameter("tls config is nil")
 	}
-	return configure(c.tlsConfig), nil
+	return configureTLS(c.tlsConfig), nil
 }
 
-// SSHClientConfig returns SSH configuration used to connect to Proxy.
-func (c *TLSConfigCreds) SSHClientConfig() (*ssh.ClientConfig, error) {
+// SSHClientConfig returns SSH configuration.
+func (c *tlsConfigCreds) SSHClientConfig() (*ssh.ClientConfig, error) {
 	return nil, trace.NotImplemented("no ssh config")
 }
 
-// LoadKeyPair is used to load credentials from files on disk.
-func LoadKeyPair(certFile string, keyFile string, caFile string) *KeyPairCreds {
-	return &KeyPairCreds{
+// LoadKeyPair is used to load Credentials from a certicate keypair on disk.
+//
+// KeyPair Credentials can only be used to connect directly to a local Teleport Auth server.
+//
+// New KeyPair files can be generated with tsh or tctl.
+//  $ tctl auth sign --format=tls --user=api-user --out=path/to/certs
```

Comment:
Is this possible? I don't see it in the CLI help or docs.

Reply:
Ah, I thought it would be `tsh login --format=tls --out=filename` but that `--format=tls` is not valid in `tsh` for some reason.
In that case, update the above sentence to not mention `tsh`